Source/Scene/Expression.js

```diff
@@ -170,6 +236,80 @@ define([
 
     // PERFORMANCE_IDEA: Have "fast path" functions that deal only with specific types
     // that we can assign if we know the types before runtime
+    Node.prototype._evaluateLessThan = function(feature) {
+        checkRelationalTypes(this);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        return left < right;
+    };
+
+    Node.prototype._evaluateLessThanOrEquals = function(feature) {
+        checkRelationalTypes(this);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        return left <= right;
+    };
+
+    Node.prototype._evaluateGreaterThan = function(feature) {
+        checkRelationalTypes(this);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        return left > right;
+    };
+
+    Node.prototype._evaluateGreaterThanOrEquals = function(feature) {
+        checkRelationalTypes(this);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        return left >= right;
+    };
+
+    Node.prototype._evaluateOr = function(feature) {
+        var left = this._left.evaluate(feature);
+        //>>includeStart('debug', pragmas.debug);
+        if (typeof(left) !== 'boolean') {
+            throw new DeveloperError('Error: Operation is undefined');
+        }
+        //>>includeEnd('debug');
+
+        // short circuit the expression
+        if (left) {
+            return true;
+        }
+
+        var right = this._right.evaluate(feature);
+        //>>includeStart('debug', pragmas.debug);
```

Comment:
This is OK with me (well, hopefully we can catch it at compile time), but there is subtle behavior here.  If the short circuit passes, it would let this error go unnoticed.  I bet JavaScript engines have the same behavior short circuiting is part of the spec.